pkg/controller/kubedirectorconfigmap/configmap.go

```diff
@@ -0,0 +1,218 @@
+// Copyright 2019 Hewlett Packard Enterprise Development LP
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+
+//     http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package kubedirectorconfigmap
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	kdv1 "github.com/bluek8s/kubedirector/pkg/apis/kubedirector/v1beta1"
+	"github.com/bluek8s/kubedirector/pkg/observer"
+	"github.com/bluek8s/kubedirector/pkg/shared"
+	"github.com/go-logr/logr"
+	corev1 "k8s.io/api/core/v1"
+)
+
+var (
+	// StatusGens is exported so that the validator can have access
+	// to the KubeDirectorConfig CR StatusGens
+	StatusGens = shared.NewStatusGens()
+)
+
+// syncConfigMap runs the reconciliation logic. It is invoked because of a
+// change in or addition of configmap instance, currently there is no
+// polling for this resource. If the configmap is not labeled
+// with key "kubedirectorcmtype" then no op
+func (r *ReconcileKubeDirectorConfigMap) syncConfigMap(
+	reqLogger logr.Logger,
+	cr *corev1.ConfigMap,
+) error {
+	// Memoize state of the incoming object.
+	oldMap, _ := observer.GetConfigMap(cr.Namespace, cr.Name)
+	hadFinalizer := shared.HasFinalizer(cr)
```

Comment:
It doesn't look like we have any need for finalizers.

Except if the configmap is removed? This code isn't currently doing anything on configmap removal (I think) but what would the desired behaviour be? Do we just notify (update) any kdclusters that have connected to the configmap?